src/MinimalDelegation.sol

```diff
@@ -126,22 +127,28 @@ contract MinimalDelegation is IERC7821, IKeyManagement, ERC1271, EIP712 {
 
     /// @dev Keyhash logic not implemented yet
     function _isValidSignature(bytes32 hash, bytes calldata signature) internal view override returns (bool) {
-        (bool isValid,) = _unwrapAndValidateSignature(hash, signature);
-        return isValid;
-    }
-
-    /// @dev Returns if the signature is valid, along with its `keyHash`.
-    /// @dev If signed with the root private key the keyHash is 0.
-    function _unwrapAndValidateSignature(bytes32 digest, bytes calldata signature)
-        internal
-        view
-        returns (bool isValid, bytes32 keyHash)
-    {
         // If the signature's length is 64 or 65, treat it like an secp256k1 signature.
         if (signature.length == 64 || signature.length == 65) {
-            return (ECDSA.recoverCalldata(digest, signature) == address(this), 0);
+            return ECDSA.recoverCalldata(hash, signature) == address(this);
         }
-        // not implemented
-        revert("Not implemented");
+        // Otherwise, treat the signature as a wrapped signature, and unwrap it before validating.
+        return _unwrapAndValidate(hash, signature);
+    }
+
+    /// @dev Returns if the wrapped signature is valid.
+    /// TODO: Implement WebAuthnP256 validation.
+    function _unwrapAndValidate(bytes32 digest, bytes calldata wrappedSignature) internal view returns (bool) {
```

Comment:
For my PR and work I think it might be better to break this up even further into an unwrap and a separate validate function, since we need to decode the wrappedSignature later to pass into _execute calls

Reply:
yup totally, i think we will have to refactor to accomadate for ur PR but figured I could pull in your changes into this before we merge this?

Reply:
im just not sure exactly how I want this to look until I play around with your refactor

Reply:
sounds good!